### DISCUSSION TITLE
关于ikuai+OpenWrt网卡配置的疑问

### DISCUSSION THREAD

**Discussion by JunwayGuo** [joyanhui/ikuai-bypass#53] [Q&A] [answered]
按照文档说明配置后，wan2，也就是OpenWrt无法走通，openclash的科学环境无法访问，（已排除OpenWrt本身的环境问题，在OpenWrt中可以ping通外网）
关于OpenWrt设置的疑问, eth0如何设置成10.1.1.1，这个不是ikuai的lan地址吗？实测中，设置成10.1.1.1，系统会自动变为192.168.X.X，文档中是不是应该设置成ikuai wan2的网关，即10.0.0.1？
不过，后续也将地址过设置成10.0.0.1，依然无法访问，
烦请作者有空看一下我贴出来的配置，是否存在问题。
![QQ20241223-233548](https://github.com/user-attachments/assets/0441a0dd-b74c-4f3e-ac3e-19516100511e)
![QQ20241223-233341](https://github.com/user-attachments/assets/517c1877-06b0-4fc9-aada-b75f76318974)
![QQ20241223-233356](https://github.com/user-attachments/assets/5016e45b-0bac-42a7-b046-ddbfe117d1e3)
![QQ20241223-233409](https://github.com/user-attachments/assets/7e2092ab-7c35-4d7d-9b46-c971edff68f1)
![QQ20241223-233425](https://github.com/user-attachments/assets/ab77ae17-5a35-457c-99dd-b57dca1e3c18)
![QQ20241223-233440](https://github.com/user-attachments/assets/20472524-1360-4512-9a45-39964fad7dca)
![QQ20241223-233454](https://github.com/user-attachments/assets/0c0e3656-e73b-439a-9308-3a14f8d2859d)

**Comment by joyanhui** [comment]
此处是文档的笔误 是应该设置成ikuai wan2的网关，即10.0.0.1，已经更新文档
可以参考图片 https://raw.githubusercontent.com/joyanhui/ikuai-bypass/refs/heads/main/assets/img.png

openwrt的lan口ip配置 看截图没问题。


至于为什么openwrt不能上网，这个和lan口理论上没有任何关系。因为openwrt自己是通过wan口上网的。具体你的原因我没看到

**Comment by JunwayGuo** [comment]
可能无没表达清楚，让您误会了。我的OpenWrt并非不能上网，OpenWrt本机是可以访问任何网络的，只是ikuai无法通过wan2访问OpenWrt，实际上通过10.0.0.0这个网段也无法相互ping通。
![QQ20241225-112003](https://github.com/user-attachments/assets/0fdf0795-8e4f-4245-9dad-56c6c48b7c0c)

**Comment by joyanhui** [comment]
这个网口物理有关系吧。爱快的wan2 是连接到op的lan口下的。wan2的ip可以手动指定 也可以由op的dhcp分配

如果都是虚拟口，注意虚拟口的配置 pve 和 ikuai的 vm 配置不同。

**Reply by JunwayGuo** [reply]
我的环境如下，4口工控机，ikuai和OpenWrt都在pve下，物理网卡都分配给了ikuai，OpenWrt虚拟2个网卡。
ikuai的wan2绑定的是物理网卡，麻烦您看看，是哪里的问题。
![QQ20241226-133858](https://github.com/user-attachments/assets/1aeecf17-cc58-4c57-845a-30d14fae5928)
![QQ20241226-133846](https://github.com/user-attachments/assets/93ee8438-0196-45b7-ac4d-12bea70eec55)
![QQ20241226-133829](https://github.com/user-attachments/assets/9d51d446-6ef8-48f3-86a3-59eeb32f003c)
![QQ20241226-133819](https://github.com/user-attachments/assets/4507f03a-0022-48e9-af42-07fe3ba0af89)

**Comment by joyanhui** [comment]
问题挪到discussion

**Comment by joyanhui** [comment, marked as answer]
我大概看了以下，我的理解是 ，你的爱快内的除了mac地址解释是5c的这个virtio网口 其他网口和op 没有连接关系。

而你把5c这个口绑定到了lan1，也是就这个口是op的wan的上游，所以op是可以上网的。

但是你op的lan口 通过什么连接到爱快？   现在实际是只接到了pve的网桥，并没有到爱快的wan2. （你爱快的wan2貌似是物理口）所以爱快自然无法访问op的lan口也就无法通过op访问到外网。


如果使用域名分流和国内ip分流（虚拟isp）你的解决方案有两个
1是 pve再新建一个虚拟网口给爱快，然后作为爱快的wan2，这个wan2的上游是op的lan口的下游。

2把op挪到爱快内的kvm里面也给两个虚拟网口

另外你4个物理网口，石沟应该是1个绑定到wan1连接光猫  剩下三个作为lan 连接内网设备
